### PR TITLE
Close the capture gate's unreachable resolution, and the third door around it

### PR DESCRIPTION
**Pull request by jerimiah797**
Found during the v0.18.0 documentation pass, while checking the release-channels
checklist item about new refusals: *"Agents retry unexplained errors, so a new
non-retryable failure has to be described as one."* Writing that up surfaced
that the failure is not only undescribed — one of the ways out of it cannot be
taken.

Release is on hold for this rather than shipping it and documenting around it.

## The gap

#152 gated `set_local_capture` on cert trust, matching `configure_system`. The
428 body offers three resolutions:

| Resolution | Reachable from MCP before this PR |
|---|---|
| `install_proxy_cert` | yes |
| `set_auto_install_cert` | yes (config / CLI) |
| `skip_cert_check` | **no** |

`LocalCaptureRequest.skip_cert_check` exists and the endpoint honours it, but
the `set_local_capture` tool's schema is `strictParams`, i.e. `.strict()`. An
agent that reads the refusal and passes the field the message names fails
schema validation before a request is made. Its remaining move is to retry a
refusal that will never clear on its own.

Practically: nobody can exercise TLS-failure paths through local capture from
an agent, and an agent that hits this mid-session has no way forward that does
not require the user to consent to a root CA — which is the outcome
`refusal_detail` was deliberately written to avoid railroading people into.

## Oversight, not a decision

Checked before assuming:

- the commit that added the gate (`0810774`) touched `server/` and `tests/`
  only, never `mcp/`
- the sole commit touching `mcp/src/tools/proxy.ts` in this cycle (`967ed42`)
  is about process names, and is unrelated
- `configure_system` has taken `skip_cert_check` since the preflight was
  written, so the parameter already existed in the tree and in the model
- #152's own verification table records `skip_cert_check: true → 200`. True,
  and tested over HTTP — the half that was never broken

The MCP tool is the one surface with no compiler and no caller to complain when
it falls behind the model. Notably `967ed42`'s subject is "in all three places",
so propagating across surfaces is already the convention here; this just missed.

## The test

Pinned as the contract, not the instance. Every handler calling
`_ensure_ca_is_trusted` is read out of the source; each must both accept
`skip_cert_check` on its body model and expose it on every MCP tool that posts
to its path. A third capture path added later fails this test rather than the
person who hits the refusal.

A parse guard asserts both known endpoints are actually found — a regex that
silently matched nothing would make the rest of the class vacuous, which is the
failure mode this file has hit before.

Mutation: removing the parameter from the tool fails
`test_every_gated_endpoint_exposes_the_parameter_to_agents`, naming both the
tool and the path.

## Also here

The tool description now states what the endpoint has done since #152 and the
description never said: **setting the list replaces it**. An agent told
"capture MyApp" sends `["MyApp"]` and silently drops the web-view defaults.
`a216ab8` made that visible in the response and the server log, and left the
tool that provokes it silent.

2337 passed, ruff clean, tsc clean.

## Summary by CodeRabbit

* **New Features**
  * Added certificate-trust checks when starting or configuring the system proxy, with an option to bypass checks for deliberate TLS testing.
  * Capture setup can install simulator certificates when configured; otherwise, guidance is provided to resolve trust issues.
  * Added optional certificate-check bypass support when enabling local capture.
  * Increased source-loading timeouts for slower devices.
* **Documentation**
  * Clarified that the capture process list is replaced entirely.
* **Bug Fixes**
  * Improved error messaging for HTTP responses versus connection failures.
* **Tests**
  * Added consistency checks for certificate-bypass options across API endpoints and tools.

### CODE REVIEW

**Comment by jerimiah797** [comment]
## Scope grew, deliberately — two more commits

Three agent reviews ran on the first commit (CodeRabbit is out of budget). Two found real defects, both mine, and one found a hole bigger than the bug this PR started with.

### `9997d81` — review fixes

- **The description promised a report the response does not carry.** I wrote "the response reports what was dropped." It does not — the response is `ProxyStatusResponse`, which has no such field. `a216ab8` says in terms that the API *logs* it and the CLI prints it, and its tests assert on `caplog`. So an agent would have been told it would be warned about something it will not be warned about: `a216ab8`'s own defect with a reassurance on top. Now points at `query_logs(source: "server")`.
- **The test's headline claim was false** for the three shapes a third gated path would actually take: a handler in `proxy_certs.py` (which already mounts the same prefix), a `@router.post(path=...)` decorator, or a cross-module `proxy._ensure_ca_is_trusted` attribute call. All three left the suite green with the bug present. Rebuilt on `test_readme_sync.py`'s existing all-files walk and moved there, which also removes a second source-scanner that would have drifted from the first.
- **The one clever trick was undefended.** Slicing the schema at a `}, async` literal is silently disarmed by the expanded handler form that `intercept.ts`, `system.ts` and `helpers.ts` already use. Now delimited by matching parens, with string literals skipped so `"(default: false)"` cannot miscount.
- **It cried wolf on three no-op refactors** — renaming `body` to `payload`, keyword-only, `Optional[X]` — each failing with "is gated but takes no typed body" about a handler that has one. The model is found by type now.
- The floor assertion moved *inside* the helper. As a sibling test it protected the suite while letting the two assertions that range over it pass trivially: with the parse empty, the run reported `1 failed, 2 passed`.

### `b4068ff` — the third door

`POST /proxy/start {system_proxy: true}` called `detect_and_configure` — the same function `configure_system` calls one line after its preflight — **with no preflight of its own**. The refusal was routable around in one MCP call:

```
configure_system_proxy  → 428
stop_proxy              → 200
start_proxy {system_proxy: true}  → 200, system proxy configured, nobody asked
```

and the tool's own workflow text recommends that sequence. The gate's docstring claimed "both paths that begin routing a device's traffic through the proxy"; there were three. The design doc's live matrix evaluated `start_proxy` as "binding a listener routes nothing" — true, and true only without the flag.

The gate runs **before** `adapter.start()`: refusing afterwards would leave the listener running behind a rejected request, and the retry would meet 409 instead of the 428 that explains itself. A bare start is still never refused — it routes nothing, and it is the call you make on the way to fixing the refusal.

`body: dict` became `StartProxyRequest`, per CONTRIBUTING and for the same `bool("false")` reason #152 hit on this path's sibling. Tested with that spelling.

**Worth noting for the review:** the parity test from the first commit went red the moment the gate landed, before I had touched the tool — a path it had never seen. That is also its limit, and the two read alike: it pins that a *gated* path offers the skip, and cannot see a path that *should* be gated and is not. This hole was found by a reviewer, not by the test.

### Mutation coverage

| | |
|---|---|
| Parity test | 9 caught (incl. the 3 shapes above and "declared but not forwarded", which survived the first fix because the destructuring still mentioned the field) |
| False alarms | 3, all gone |
| `start_proxy` gate | 5 caught: removed, moved after `start()`, applied unconditionally, always-skip, field dropped from the model |

2343 passed, ruff clean, tsc clean.

### Still open — the third hole, not in this PR

`quern enable-local-capture` writes `config.json` and restarts the server; the lifespan builds the adapter and starts routing with no trust check on that path. Different fix location and different shape (a gate that cannot refuse, since failing to boot the server over a cert would be worse than the problem). Coming next, before the release.

**Comment by jerimiah797** [comment]
@coderabbitai review

**Comment by jerimiah797** [comment]
@coderabbitai review

File: tests/test_readme_sync.py [hunk 1]
```diff
@@ -482,3 +482,270 @@ def test_help_is_a_command_not_only_a_flag():
...
+    Parens inside a `.describe()` string are common ("(default: false)"), so a
+    naive counter miscounts on prose. Strings are skipped wholesale.
+    """
+    depth = 0
+    i = open_idx
+    while i < len(src):
+        c = src[i]
+        if c in "\"'`":
+            i = _skip_strings(src, i)
+            continue
```
[the remark this answers is not in the document]
Comment (jerimiah797) [reply tests/test_readme_sync.py:665]: Fixed in 4b3621c. Real, and not hypothetical on this branch — the previous commit added a comment inside the `system_proxy` zod block, and `intercept.ts`, `system.ts` and `helpers.ts` already carry comments in that position. It survived only because none of them contains a lone apostrophe.

`_balanced` now skips `//` and `/* */`, checking for a comment before a quote so that a `//` inside a string is still a string.

Worth separating the two failure modes you named: the no-boundary case would have failed loudly, because `test_the_schema_parse_finds_a_real_schema_for_every_tool` rejects an empty schema slice. The wrong-boundary case would not have, and that is the one that matters — it silently changes which text the parity assertions read. Three tests now cover the apostrophe-in-line-comment, block-comment, and the original paren-in-`describe()` case. Removing the comment handling fails the first.

File: server/main.py [hunk 2]
```diff
@@ -437,6 +450,16 @@ async def _warmup_devices():
...
     # Note: does NOT kill xcodebuild processes — they persist across restarts
+    warmup = getattr(app.state, "_warmup_task", None)
+    if warmup and not warmup.done():
+        # It used to only warm caches, where an abrupt teardown cost nothing.
+        # It now installs a CA and then records that it did, and losing the
+        # loop between those two steps leaves the TrustStore holding a
+        # certificate that cert-state.json says is absent.
+        warmup.cancel()
+        with contextlib.suppress(asyncio.CancelledError):
+            await warmup
```
[the remark this answers is not in the document]
Comment (jerimiah797) [reply server/main.py:461]: Fixed in 4b3621c, by shielding the install.

You are right about the mechanism: cancelling the await does not terminate a subprocess that has already run, and `install_cert` shells out to simctl and *then* writes the record, so a shutdown landing between those leaves the TrustStore holding a CA that `cert-state.json` calls absent.

One qualification on the impact, which is why this is a small fix rather than an urgent one: that state is self-correcting. ADR 1 deleted the trust cache in this same release, so every caller now asks the device rather than reading the record, and the next check repairs it. A stale `false` costs one redundant install; the dangerous direction is a stale `true`, which this cannot produce. But 'it heals later' is a reason not to panic, not a reason to write it deliberately.

The test drives it rather than asserting on the source — starts the task, cancels mid-install, asserts the record was still written. Removing the shield fails it.
